package/kedro_viz/server.py

```diff
@@ -213,31 +261,62 @@ def commands():
     type=click.Path(dir_okay=False, writable=True),
     help="Path to save the pipeline JSON file",
 )
-def viz(host, port, browser, load_file, save_file):
+@click.option(
+    "--pipeline",
+    type=str,
+    default=None,
+    help="Name of the modular pipeline to run. "
+    "If not set, the default pipeline is visualized",
+)
+@click.option(
+    "--env",
+    "-e",
+    type=str,
+    default=None,
+    multiple=False,
+    envvar="KEDRO_ENV",
+    help="Kedro configuration environment. If not specified, "
+    "catalog config in `local` will be used",
+)
+def viz(host, port, browser, load_file, save_file, pipeline, env):
     """Visualize the pipeline using kedroviz."""
-    _call_viz(host, port, browser, load_file, save_file)
-
-
-def _call_viz(host=None, port=None, browser=None, load_file=None, save_file=None):
+    try:
+        _call_viz(host, port, browser, load_file, save_file, pipeline, env)
+    except KedroCliError:
+        raise
+    except Exception as ex:
+        raise KedroCliError(str(ex))
+
+
+# pylint: disable=too-many-arguments
+def _call_viz(
+    host=None,
+    port=None,
+    browser=None,
+    load_file=None,
+    save_file=None,
+    pipeline_name=None,
+    env=None,
+):
     global data  # pylint: disable=global-statement,invalid-name
 
     if load_file:
-        data = json.loads(Path(load_file).read_text())
-        for key in ["nodes", "edges", "tags"]:
-            if key not in data:
-                click.echo("Invalid file, top level key '{}' not found.".format(key))
-                sys.exit(1)
+        data = _load_from_file(load_file)
     else:
-        try:
-            pipeline = get_project_context("create_pipeline")()
-            catalog = get_project_context("create_catalog")(None)
-        except KedroContextError:
-            raise KedroCliError(
-                "Could not find a Kedro project root. "
-                "You can run `kedro viz` by either providing `--load-file` flag with a JSON file "
-                "path for your pipeline, or if the current working directory is "
-                "the root of a Kedro project."
-            )
+        if match(kedro.__version__, ">=0.15.0"):
+            from kedro.context import KedroContextError
+
+            try:
+                context = get_project_context("context", env=env)
+                pipeline = _get_pipeline_from_context(context, pipeline_name)
+            except KedroContextError:
+                raise KedroCliError(ERROR_PROJECT_ROOT)
+            catalog = context.catalog
+
+        else:
+            # Kedro 0.14.*
+            pipeline, catalog = _get_pipeline_catalog_from_kedro14(env, pipeline_name)
+
         data = format_pipeline_data(pipeline, catalog)
```

Comment:
Why is the catalog needed to visualise the pipeline?

Reply:
@idanov It's not necessary at the moment, but catalog was added in https://github.com/quantumblacklabs/kedro-viz/pull/72

This is Gordon's comment. 

![Screenshot 2020-02-06 at 14 02 26](https://user-images.githubusercontent.com/8097799/73943806-52aef900-48e9-11ea-8295-d140b320935b.png)